app/code/Magento/Integration/Controller/Adminhtml/Integration/LoginSuccessCallback.php

```diff
@@ -15,6 +15,6 @@ class LoginSuccessCallback extends \Magento\Integration\Controller\Adminhtml\Int
      */
     public function execute()
     {
-        $this->getResponse()->setBody('<script>setTimeout("self.close()",1000);</script>');
+        $this->getResponse()->setBody('<script>setTimeout(function () { self.close(); }, 1000);</script>');
```

Comment:
Hello @jonathanKingston. You can don't use wrapper function. For that, you should put method "close" to setTimeout without execution. (setTimeout(self.close, 1000))

Reply:
This code isn't clear where `self.close` is from, executing it this way prevents it being a string also. This would also make linting simpler in future if you wanted to prevent this in future.

Reply:
ok, I understood the problem, ok, the solution is ok. Also, you can use "bind" function to wrapping self.case method.

Reply:
Cool will do that as I have seen this elsewhere. I'm just checking over all other uses of setTimeout now.